rust/routee-compass-core/src/util/multiset.rs

```diff
@@ -9,13 +9,13 @@ where
     T: Clone + Copy,
 {
     fn from(sets: &'a Vec<Vec<T>>) -> Self {
-        let final_pos: Vec<usize> = sets.into_iter().map(|v| v.len() - 1).collect();
-        let pos: Option<Vec<usize>> = Some(vec![0 as usize; sets.len()]);
-        return MultiSet {
+        let final_pos: Vec<usize> = sets.iter().map(|v| v.len() - 1).collect();
+        let pos: Option<Vec<usize>> = Some(vec![0_usize; sets.len()]);
```

Comment:
this looks like a typo. is `0_usize` valid rust code?

Reply:
This was one of the auto clippy fixes ([here's the lint doc](https://rust-lang.github.io/rust-clippy/master/index.html#/unnecessary_cast)) and I didn't know this but apparently it is [valid rust code](https://doc.rust-lang.org/rust-by-example/types/literals.html#literals). But just putting in `0` would also work since we have an explicit type hint and so maybe I just do that